File: src/main/java/ar/edu/um/tesoreria/rest/model/Comprobante.java
/**
 * 
 */
package ar.edu.um.tesoreria.rest.model;

import java.io.Serializable;

import jakarta.persistence.Column;
import jakarta.persistence.Entity;
import jakarta.persistence.Id;
import jakarta.persistence.Table;

import lombok.AllArgsConstructor;
import lombok.Data;
import lombok.EqualsAndHashCode;
import lombok.NoArgsConstructor;

/**
 * @author daniel
 *
 */
@Data
@Entity
@Table(name = "tiposcomprob")
@EqualsAndHashCode(callSuper = false)
@NoArgsConstructor
@AllArgsConstructor
public class Comprobante extends Auditable implements Serializable {
	/**
	* 
	*/
	private static final long serialVersionUID = -67735353263838005L;

	@Id
	@Column(name = "tco_id")
	private Integer comprobanteId;

	@Column(name = "tco_descripcion")
	private String descripcion;

	@Column(name = "tco_modulo")
	private Integer tipoTransaccionId;

	@Column(name = "tco_opago")
	private Byte ordenPago = 0;

	@Column(name = "tco_aplicapend")
	private Byte aplicaPendiente = 0;

	@Column(name = "tco_ctacte")
	private Byte cuentaCorriente = 0;

	@Column(name = "tco_debita")
	private Byte debita = 0;

	@Column(name = "tco_diasvigencia")
	private Integer diasVigencia = 0;

}

File: src/main/java/ar/edu/um/tesoreria/rest/kotlin/model/Comprobante.kt
package ar.edu.um.tesoreria.rest.kotlin.model

import ar.edu.um.tesoreria.rest.model.Auditable
import jakarta.persistence.Column
import jakarta.persistence.Entity
import jakarta.persistence.Id
import jakarta.persistence.Table

@Entity
@Table(name = "tiposcomprob")
data class Comprobante(

    @Id
    @Column(name = "tco_id")
    var comprobanteId: Int? = null,

    @Column(name = "tco_descripcion")
    var descripcion: String = "",

    @Column(name = "tco_modulo")
    var tipoTransaccionId: Int? = null,

    @Column(name = "tco_opago")
    var ordenPago: Byte = 0,

    @Column(name = "tco_aplicapend")
    var aplicaPendiente: Byte = 0,

    @Column(name = "tco_ctacte")
    var cuentaCorriente: Byte = 0,

    @Column(name = "tco_debita")
    var debita: Byte = 0,

    @Column(name = "tco_diasvigencia")
    var diasVigencia: Long = 0,

    var facturacionElectronica: Byte = 0,

    var tipoAfip: Int? = null,

    var puntoVenta: Int? = null,

    ) : Auditable()
